src/hotspot/share/classfile/stackMapTableFormat.hpp

```diff
@@ -549,7 +549,7 @@ class append_frame : public stack_map_frame {
 
   static u1 appends_to_frame_type(int appends) {
     assert(appends > 0 && appends < 4, "Invalid append amount");
-    return 251 + appends;
+    return (u1)(251 + appends);
```

Comment:
This should use `checked_cast<u1>`

Reply:
This one clearly is < u1 since it checks the value for appends.

Reply:
I would use static_cast to be consistent with `chops_to_frame_type`. That way the reader of this code has less metal exercise to do.

Reply:
I've struggled with the raw cast versus checked_cast usage in this PR - I could not determine what was guiding the usage of each type of cast. Shouldn't it always be a checked cast? Or does the preceding range check validate the use of a raw cast (so we don't range check twice)?

Reply:
To me, the two lines serves two purposes:

```
assert(appends > 0 && appends < 4, "Invalid append amount");
```
This is the input expected by this function.

```
    return static_cast<u1>(251 + appends);
```

This is a mechanical check that the code is actually correctly written (i.e., you didn't mess up and wrote (253 + appends). That way the reader doesn't have do the math in order to trust this code.

Reply:
True - the assert is checking a precondition; the checked_cast would be checking a postcondition.